### PR TITLE
Ticket2469 ieg system

### PR DESCRIPTION
**Pull request by Tom-Willemsen**
### Description of work

IOC for the IEG gas exchange system

### To test

2469

### Acceptance criteria

- [x] IOC implements the command set as described in the documentation (linked in the original ticket)
- [x] IOC communicates with the emulator as expected
- [x] IOC test framework tests exist and pass

- [x] Calibration can be defined in `globals.txt` using the following lines
```
IEG_01__CALIBRATION_A=0.2
IEG_01__CALIBRATION_B=5
```
- [x] If you have defined a calibration and then set the sample_pressure on the emulator via the backdoor, the value is converted correctly into `IEG_01:PRESSURE` using the standard linear equation `y = ax + b` with a and b equal to the values defined in globals.txt
- [x] If the pressure is greater than 350 mBar it is displayed on the GUI as `> 350 mBar`

---

#### Code Review

- [x] Is the code of an acceptable quality?
- [x] Are the PVs named according to the [naming standards](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/PV-Naming).
- [x] Have suitable [disable records](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Disable-records) been added?
- [x] Have suitable [simulation records](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Record-Simulation) been added?
- [x] Have the [finishing touches](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/IOC-Finishing-Touches) been applied?
- [x] Is there an [emulator](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Emulating-Devices) for the device?
- [x] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev). If so, do they describe the changes appropriately?
- [x] If an OPI has been modified, does it conform to the [style guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/OPI-Creation)? There is a script called `check_opi_format.py` to help with this.

### Functional Tests

- [x] Do changes function as described? Add comments below that describe the tests performed.
- [x] Does the IOC respond correctly both in full and simulation mode, where it's possible to test both?
- [ ] If there are multiple _0n IOCs, do they run correctly?
- [x] Ensure that the log files do not contain undefined macros, ie serach for `macLib: macro` full text is `macLib: macro XXXXX is undefined (expanding string XXXX)`

### Final steps

- [ ] Reviewer has updated the submodule in the main EPICS repo? See **Reviewing work for the subModules of EPICS** in the [Git workflow](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Git-workflow) page for details.
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

**Comment by John-Holt-Tessella** [comment]
As well as comments in code:
 - [ ] add an alarm pv

File: IEG/IEG-IOC-01App/Db/ieg.db [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+	info(INTEREST, "HIGH")
+	info(archive, "VAL")
+}
+
+record(ai, "$(P)PRESSURE:BUFFER:HIGH")
+{
+	field(SCAN, "Passive")
+	field(DTYP, "Soft Channel")
+	field(DESC, "Non-zero if buffer pressure is too high")
```
Comment (John-Holt-Tessella) [inline IEG/IEG-IOC-01App/Db/ieg.db]: Update comment (0 if out of range)
Add operating range

File: IEG/IEG-IOC-01App/Db/ieg.db [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+	field(EGU, "")
+	info(INTEREST, "HIGH")
+	info(archive, "VAL")
+}
+
+record(ai, "$(P)PRESSURE:HIGH")
+{
+	field(SCAN, "Passive")
+	field(DTYP, "Soft Channel")
+	field(DESC, "Non-zero if pressure is too high")
```
Comment (John-Holt-Tessella) [inline IEG/IEG-IOC-01App/Db/ieg.db]: Add operating range

File: IEG/IEG-IOC-01App/Db/ieg.db [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+	field(EGU, "")
+	info(INTEREST, "HIGH")
+	info(archive, "VAL")
+}
+
+record(ai, "$(P)PRESSURE:LOW")
+{
+	field(SCAN, "Passive")
+	field(DTYP, "Soft Channel")
```
Comment (John-Holt-Tessella) [inline IEG/IEG-IOC-01App/Db/ieg.db]: Add operating range

File: IEG/IEG-IOC-01App/Db/ieg.db [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:MODE")
+    field(SDIS, "$(P)DISABLE")    
+    
+    info(INTEREST, "HIGH")
+    info(archive, "VAL")
+}
+
+record(mbbiDirect, "$(P)VALVESTATE")
```
Comment (AdrianPotter) [inline IEG/IEG-IOC-01App/Db/ieg.db]: https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/PV-Naming

Please could this be `VALVE:STAT`

**Pull request merged**